src/ztc/api/utils/viewsets.py

```diff
@@ -128,8 +140,10 @@ def extract_relevant_m2m(serializer, m2m_fields: list, action: str, date=None):
         for query_object in data:
             valid_urls = []
             for m2m_url in query_object[m2m_field]:
-                uuid_from_url = uuid.UUID(m2m_url.rsplit("/", 1)[1]).hex
-
+                if m2m_field == "gerelateerde_zaaktypen":
+                    uuid_from_url = uuid.UUID(m2m_url["zaaktype"].rsplit("/", 1)[1]).hex
+                else:
+                    uuid_from_url = uuid.UUID(m2m_url.rsplit("/", 1)[1]).hex
                 valid_m2m = get_valid_m2m_objects(m2m_field, uuid_from_url, date)
 
                 if valid_m2m:
```

Comment:
je append hier een url die mogelijk geen url is want je overschrijft nergens m2m_url als het een dict is.

Ik stel dit voor:

                # if the instance is a dict it is a gerelateerdeZaak which means we need the zaaktype
                if isinstance(m2m_url, dict):
                    m2m_url = m2m_url['zaaktype']

                uuid_from_url = uuid.UUID(m2m_url.rsplit("/", 1)[1]).hex

we kunnen kijken of we mogelijk het zaaktype nog met een loop willen doen tot we een url hebben want het zou best kunnen dat er iets bijkomt

Reply:
wellicht ook even kijken of het url is zodat we alle hardcoded strings zoals "zaaktype" kwijt zijn.

Reply:
> wellicht ook even kijken of het url is zodat we alle hardcoded strings zoals "zaaktype" kwijt zijn.

Wat bedoel je hier precies mee? `zaaktype` is toch een attribuut wat uitgelezen wordt?